src/Common/ProxyConfigurationResolverProvider.cpp

```diff
@@ -0,0 +1,103 @@
+#include <Common/ProxyConfigurationResolverProvider.h>
+
+#include <Common/EnvironmentProxyConfigurationResolver.h>
+#include <Common/ProxyListConfigurationResolver.h>
+#include <Common/RemoteProxyConfigurationResolver.h>
+#include <Common/Exception.h>
+#include <Common/StringUtils/StringUtils.h>
+#include <Common/logger_useful.h>
+
+#include <Interpreters/Context.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+}
+
+std::shared_ptr<ProxyConfigurationResolver> ProxyConfigurationResolverProvider::get()
+{
+    if (auto context = Context::getGlobalContextInstance())
```

Comment:
I am not so confident about this anymore. Relying on global state usually bites us back in the future. I'll consider modifying a couple of external APIs to take the context / config back in.